kglm/data/dataset_readers/enhanced_wikitext.py

```diff
@@ -193,6 +197,12 @@ def text_to_instance(self, data: Dict[str, Any]) -> Instance:  # pylint: disable
                     shortlist.append(entity_id)
                 shortlist_ind = reverse_shortlist[entity_id]
 
+                # Collect the relations and their parents
+                relations.extend(annotation["relation"])
+                parent_ids.extend(annotation["parent_id"])
```

Comment:
With this approach how can you determine which annotation each relation and parent id come from?

Reply:
right, this was why i was asking about the linking the (entity, relation, parentid) triples last night